data_model/src/permissions.rs

```diff
@@ -148,13 +178,41 @@ impl PermissionToken {
         self.params.get(name)
     }
 
-    /// Get an iterator over parameters of the `PermissionToken`
+    /// Get an iterator over parameters of the [`PermissionToken`].
+    ///
+    /// Values returned from the iterator are guaranteed to be in the alphabetical order.
     #[inline]
     pub fn params(&self) -> impl ExactSizeIterator<Item = (&Name, &Value)> {
         self.params.iter()
     }
 }
 
+/// Trait to identify [`ValueKind`] of a type which can be used as a [`PermissionToken`] parameter.
+///
+/// On higher level, all permission token parameters have [`Value`] type, but for now we allow
+/// to define builtin permission tokens with stronger types.
+/// This trait is used to retrieve the [`kind`](`ValueKind`) of a [`Value`] which can be constructed
+/// from given parameter.
+///
+/// Will be removed as well as builtin permission tokens and validators
+/// when *runtime validators* and *runtime permissions* will be properly implemented.
+pub trait PermissionTokenValueTrait: Into<Value> {
+    /// Kind of the [`Value`] which the implementing type can be converted to.
```

Comment:
```suggestion
    /// The kind of the [`Value`] which the implementing type can be converted to.
```

"kind of" has its own separate meaning, so it's a bit confusing